### DISCUSSION TITLE
Default compact attribution control should stay expanded if container >= 640px ?

### DISCUSSION THREAD

**Discussion by Al-4SW** [maplibre/maplibre-gl-js#4749] [Q&A] [answered]
Hi,

The API docs for 'AttributionControlOptions' say that the default compact control will always collapse when moving the map, but then later say the default is a responsive attribution that collapses when the user moves the map on maps <640px wide, and attributions shouldn't be collapsed if it can comfortably fit on the map.

Currently the default attribution collapses on move for any container width. Should this be staying open when > 640px?

**Comment by acalcutt** [comment, marked as answer]
In https://github.com/maplibre/maplibre-gl-js/pull/3618 the attribution was made compact by default.  In my PR https://github.com/maplibre/maplibre-gl-js/pull/795 before that, the compact mode was made to default to open

It looks like to get the old behavior, you would need to set the compact option to 'undefined', which makes it default to the <640px wide compact, >640 not compact mode. Something like https://stackblitz.com/edit/web-platform-tqhkxm?file=index.html

With compact mode set to true, the attribution is always open by default, like
![image](https://github.com/user-attachments/assets/b627576c-477f-4281-95d9-39d761e3c50d)
but can also be collapsed (and collapses on move)

With compact mode set to false, the attribution always shows like
![image](https://github.com/user-attachments/assets/fb4de88c-2eb8-4d9d-a722-094c575c6fd4)
but can't be collapsed

With compact mode set to 'undefined', the old behavior happens. Under 640px it looks like 
![image](https://github.com/user-attachments/assets/f1064b10-13f9-4f38-9825-f39a82e7f7c6)
(with a collapse button that closes on move)
While over 640px it looks like 
![image](https://github.com/user-attachments/assets/5370933b-ebb2-471c-a197-2a570d4b7ecf)
(with no collapse button)

**Comment by acalcutt** [comment]
So the first part which says " the default compact control will always collapse when moving the map" seems correct

But the second part you mentioned "the default is a responsive attribution that collapses when the user moves the map on maps <640px wide, and attributions shouldn't be collapsed if it can comfortably fit on the map." is outdated and was the old behavior. This behavior is seen by setting compact to 'undefined' now like in my example at https://stackblitz.com/edit/web-platform-tqhkxm?file=index.html
![heatmap test (latest) - StackBlitz - Google Chrome 2024-09-23 18-53-13 (2)](https://github.com/user-attachments/assets/551df1ba-dde7-40b9-bf93-da5aab6a301e)

**Reply by Al-4SW** [reply]
Thanks acalcutt for the detailed response, you have helped me understand what the docs are referring to now! 

The section around "_compact should only be used to modify default attribution when map size makes it impossible to fit default attribution_ " now also makes more sense to me,  understanding that it refers to when compact mode used to default to closed.

It does seem a shame to have to revert back to the old expanded control to keep the attributions on, I much prefer the look of the floating compact control.

**Reply by Al-4SW** [reply]
I took a look into attribution_control.ts in the source, a  minor change to updateCompactMinimize() would re-introduce similar behaviour with the compact control:

![Peek 2024-09-24 00-58](https://github.com/user-attachments/assets/4dc27133-bc3a-4605-8d53-206d193361b7)

I just included && this._map.getCanvasContainer().offsetWidth <= 640 in the condition to minimize. 

I don't know if this would be a welcomed change/PR? It feels like better behavior to me, the docs do say after all "Attribution should not be collapsed if it can comfortably fit on the map"

**Reply by acalcutt** [reply]
I think that looks ok, but my intent was to make compact mode only show initially to comply with OSM license. Compact mode is meant to take up less room, so I'm not sure it should stay open.

In my own site I set compact to false, since I didn't really like how much the compact mode stands out.

**Reply by acalcutt** [reply]
Always worth a shot with a PR though. there has still been some discussion around this default open behavior in https://github.com/maplibre/maplibre-gl-js/pull/4526 , which is almost the opposite of what you are asking (not showing  attribution open by default)
